collector/exporter/otelarrowexporter/internal/arrow/exporter.go

```diff
@@ -127,40 +146,52 @@ func NewExporter(
 // Start creates the background context used by all streams and starts
 // a stream controller, which initializes the initial set of streams.
 func (e *Exporter) Start(ctx context.Context) error {
-	ctx, cancel := context.WithCancel(ctx)
+	// this is the background context
+	ctx, e.doneCancel = newDoneCancel(ctx)
 
-	e.cancel = cancel
+	// Starting N+1 goroutines
 	e.wg.Add(1)
-	e.ready = newStreamPrioritizer(ctx, e.numStreams)
 
-	go e.runStreamController(ctx)
+	// this is the downgradeable context
+	downCtx, downDc := newDoneCancel(ctx)
+
+	var sws []*streamWorkState
+	e.ready, sws = newStreamPrioritizer(downDc, e.prioritizerName, e.numStreams)
+
+	for _, ws := range sws {
+		e.startArrowStream(ctx, ws)
```

Comment:
Thanks for describing the three types of context in the readme. Should this pass in the downCtx instead of the original exporter parent context (like what you do for restarts in runStreamController L194)?

Reply:
Thanks-- yes, the README says what I meant.

Reply:
The reason this isn't a detectable condition now relates to the open TODO about downgrade. The current downgrade logic waits for all of the initial contexts to finish before it can happen.